rosys/hardware/gnss.py

```diff
@@ -170,7 +172,12 @@ async def _run(self) -> None:
             self.log.debug(sentence)
             parts = sentence.split(',')
             try:
-                timestamp = parts[{'$GPGGA': 1, '$GPGST': 1, '$PSSN': 2}[parts[0]]]
+                nema_timestamp = parts[{'$GPGGA': 1, '$GPGST': 1, '$PSSN': 2}[parts[0]]]
+                today = datetime.now().date()
+                time_obj = datetime.strptime(nema_timestamp, '%H%M%S.%f').time()
+                utc_time = datetime.combine(today, time_obj).replace(tzinfo=timezone.utc)
+                timestamp = utc_time.timestamp()
+                self.log.debug('timestamp diff = %s', round(timestamp - rosys_time, 3))
```

Comment:
I think it would be more correct and more robust (around midnight) to compare times mod 24 hours instead of datetimes. But since it's only used for logging, having a 24-hour difference from time to time is _ok_ I guess...

Reply:
Oh, very good. I added the 24h code because its also good to warn if time differs to much.